modules/dynamic/src/smithy4s/dynamic/internals/DynamicLambdas.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.dynamic.internals
+
+import smithy4s.schema.Alt
+import smithy4s.schema.Schema
+
+/**
+  * A bunch of hash-code stable lambdas that are less likely to break memoization
+  * that may be able to SchemaVisitors.
+  */
+private[internals] object DynamicLambdas {
```

Comment:
Right, so this is a bit of a side-thing, but I wanted to ensure that several instances of the "same" dynamic schemas were "equal" when facing caching. To that end, we make all accessors serialisable, to make the natural `.hashCode` and `.equals` on `Schema` behave like we intuitively think they should.